tensorflow/python/training/initial_learning_rate_decay.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Various learning rate decay functions."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import constant_op
+
+
+def exponential_time_decay(initial_lr, epoch, decay_rate, name=None):
+  """Applies exponential time decay to the initial learning rate.
+
+  When training a model, it is often recommended to lower the learning rate as
+  the training progresses.  This function applies an exponential decay function
+  to a provided initial learning rate.  It requires an `epoch` value to compute
+  the decayed learning rate.  You can just pass a TensorFlow variable that you
+  increment at each training step.
+
+  The function returns the decayed learning rate.  It is computed as:
+
+  ```python
+  decayed_learning_rate = initial_lr * exp(-decay_rate * epoch)
+  ```
+
+  Example: decay exponetially with a base of 0.96:
+
+  ```python
+  ...
+  epoch = tf.Variable(0, trainable=False)
+  initial_lr = 0.1
+  k = 0.5
+  learning_rate = tf.train.exponential_time_decay(initial_lr, epoch, k)
+
+  # Passing epoch to minimize() will increment it at each step.
+  learning_step = (
+      tf.GradientDescentOptimizer(learning_rate)
+      .minimize(...my loss..., global_step=epoch)
+  )
+  ```
+
+  Args:
+    initial_lr: A scalar `float32` or `float64` `Tensor` or a
+      Python number.  The initial learning rate.
+    epoch: A Python number.
+      Global step to use for the decay computation.  Must not be negative.
+    decay_rate: A Python number.  The decay rate.
+    name: String.  Optional name of the operation.  Defaults to
+      'ExponentialTimeDecay'
+
+  Returns:
+    A scalar `Tensor` of the same type as `learning_rate`.  The decayed
+    learning rate.
+  """
+  with ops.op_scope([initial_lr, epoch, decay_rate],
+                    name, "ExponentialTimeDecay") as name:
+    initial_lr = ops.convert_to_tensor(initial_lr, name="learning_rate")
+    decay_rate = math_ops.cast(decay_rate, initial_lr.dtype)
+    exponent = math_ops.exp(math_ops.mul(math_ops.neg(decay_rate), epoch))
```

Comment:
after more closely looking at the differences, I realize that all of these functions have a 'staircase' effect: they only drop after each epoch.

I think it might be worth looking at how it is done in exponential_decay:

- You always pass the global_step and the 'decay steps' (essentially the number of steps in an epoch)
- If staircase is True, then you take the floor of global_step / decay_steps when computing the equivalent of 'epoch' here
- If staircase is False, then you use global_step / decay_steps as a floating point value, so you have smooth behavior.  That way all of the interfaces to the learning rate functions are basically identical, they each take an initial learning rate, an global step value, and a decay_steps. 

In fact, where possible, it might be preferable to keep the signature (the names of the variables) the same as in exponential_decay.  Does this seem reasonable?